ggshield/cmd/sca/scan.py

```diff
@@ -183,3 +214,25 @@ def get_sca_scan_all_filepaths(
     # Only sca_files field is useful in the case of a full_scan,
     # all the potential files already exist in `all_filepaths`
     return response.sca_files
+
+
+def sca_scan_diff(
+    ctx: click.Context,
+    directory: Path,
+    ref: str,
+    include_staged: bool,
+) -> SCAScanDiffOutput:
+    client = SCAClient(ctx.obj["client"])
+    current_ref = INDEX_REF if include_staged else "HEAD"
```

Comment:
If `current_ref==ref`, we could avoid the scan here I think.

Reply:
I don't think so, that's not really an error from the code point of view. We may print a warning message like "sca scan diff comparing the current state to itself: nothing to do." (message should be better ! )